src/test/java/task3/UrlValidatorTest.java

```diff
@@ -0,0 +1,45 @@
+package task3;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.regex.*;
+
+public class UrlValidatorTest {
+    private static final String URL_REGEX = "^(http|https)://[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}(?:/[a-zA-Z0-9._%+-]+)*$";
```

Comment:
w swoich testach stworzyłeś implementacje od nowa ;) no kompletnie nie o to w testach chodzi. Tak jak napisałem w poprzednim komentarzu. Powinieneś mieć dedykowaną metodę do walidacji np. isPeselValid(String pesel) i tą metodę powinieneś tutaj testować. A ty teraz zrobiłeś w main jakiś mini program plus kopiuj wklej do testów :) no tak się nie robi